app/javascript/controllers/chart_controller.js

```diff
@@ -141,6 +112,30 @@ export default class extends Controller {
     ]
   }
 
+  modifyDates() {
```

Comment:
Naming functions and variables is more important than you might think. I think a better name might by `formatDates()`.